src/protocol.cpp

```diff
@@ -238,7 +238,7 @@ CInv::CInv(const std::string& strType, const uint256& hashIn)
         }
     }
     if (i == ARRAYLEN(ppszTypeName))
-        LogPrint("net", "CInv::CInv(string, uint256) : unknown type '%s'\n", strType);
+        throw std::out_of_range(strprintf("CInv::CInv(string, uint256): unknown type '%s'", strType));
```

Comment:
I would also agree but I prefer to have code aligned with bitcoin as much as possible :) and for that part this PR simply make things as they were before my previous "fix" https://github.com/bitcoin/bitcoin/blob/0.12/src/protocol.cpp#L172 https://github.com/bitcoin/bitcoin/blob/0.12/src/protocol.cpp#L189